File: bond/PHP/stage1.php
/// python-bond PHP interface loader
/// NOTE: use /// for comments *only*, as this code is transformed into a
///       single line to be injected into the interpreter *without parsing*.

echo "STAGE2\n";
$__PY_BOND_STDIN = fopen("php://stdin", "r");
$__PY_BOND_STAGE2 = json_decode(rtrim(fgets($__PY_BOND_STDIN)));
eval($__PY_BOND_STAGE2->code);
call_user_func_array($__PY_BOND_STAGE2->func, $__PY_BOND_STAGE2->args);

File: bond/Python/stage1.py
### python-bond Python interface loader
### NOTE: use ### for comments *only*, as this code is reduced as much as
###       possible to be injected into the interpreter *without parsing*.

def __PY_BOND_stage1():
    import sys
    sys.stdout.write("STAGE2\n")
    sys.stdout.flush();
    line = sys.stdin.readline().rstrip()
    stage2 = eval(line)
    exec(stage2['code'], globals())
    eval(stage2['func'], globals())(*stage2['args'])

__PY_BOND_stage1()
